### PR TITLE
from_entr constructor

### LINKED ISSUE #231: Merge in "from_entr" constructor from entr fork.
Source: entralliance/OpenOA:more_flexible_entr_connector
Destination: NREL/OpenOA:develop_v3

- [x] Either merge in whole entr toolkit module (and add tests for it to the NREL/OpenOA CI), or split this into a separate python package with its own tests in the entralliance org.
- [ ] Merge the "PlantData.from_entr" stub into NREL/OpenOA plantdata. No tests are needed for this stub.

### PR DESCRIPTION
**Pull request by jordanperr**
This pull request merges in the from_entr constructor in plant.py. It also includes several small tweaks to plant.py which had to be made in the entr fork in order to get things to work.

- Added from_entr to plant.py
- Fixed plotting function so annotated percentage is in correct order of magnitude and the title is included.

Will close #231

### CODE REVIEW

File: openoa/plant.py [hunk 1]
```diff
@@ -1259,7 +1259,7 @@ def __attrs_post_init__(self):
 
         if self.scada is not None:
             self.calculate_turbine_energy()
-
+
```
Comment (RHammond2) [inline openoa/plant.py:1262]: Looks like the pre-commit workflow didn't run based on this extra whitespace, could you just make sure that it runs and passes?

File: openoa/utils/plot.py [hunk 2]
```diff
@@ -1287,6 +1287,7 @@ def plot_distributions(
     figure_kwargs: dict = {},
     plot_kwargs: dict = {},
     annotate_kwargs: dict = {},
+    title: str=None,
```
Comment (RHammond2) [inline openoa/utils/plot.py:1290]: Should be `title: str | None = None`. I know we're not technically doing any static type checking, but could you update this?

File: openoa/utils/plot.py [hunk 3]
```diff
@@ -1340,7 +1342,7 @@ def plot_distributions(
             **annotate_kwargs,
         )
         ax.annotate(
-            f"Uncentainty = {vals.std() / vals.mean():.1f}",
+            f"Uncentainty = {(vals.std() / vals.mean())*100:.1f}%",
```
Comment (RHammond2) [inline openoa/utils/plot.py:1345]: Nice adjustment!

File: setup.py [hunk 4]
```diff
@@ -44,7 +44,6 @@
         "sphinxcontrib-bibtex",
         "myst-parser",
     ],
-    "entr": ["pyhive"],
```
Comment (RHammond2) [inline setup.py:47]: Looks like nothing is in here to include the ENTR package, is that intentional?

File: openoa/plant.py [hunk 5]
```diff
@@ -2143,61 +2141,15 @@ def nearest_tower(self, id: str) -> str:
             self.calculate_nearest_neighbor()
         return self.asset.loc[id, "nearest_tower_id"].values[0]
 
+    @classmethod
+    def from_entr(cls, *args, **kwargs):
+        from entr.plantdata import from_entr
```
Comment (RHammond2) [inline openoa/plant.py:2146]: Should this be in a try/except clause and inform users to run `pip install "openoa[entr]"`?

**Review by RHammond2 — CHANGES_REQUESTED** [review]
Pretty straightforward to me. I just had a couple really small changes in there, otherwise this all looks good to me.

File: openoa/plant.py (hunk 1 above)
Comment (jordanperr) [reply openoa/plant.py]: I ran the precommit workflow.

File: openoa/utils/plot.py (hunk 2 above)
Comment (jordanperr) [reply openoa/utils/plot.py]: I made this change.

File: setup.py (hunk 4 above)
Comment (jordanperr) [reply setup.py:47]: I added a try/catch to the entr import as requested.

File: openoa/plant.py (hunk 5 above)
Comment (jordanperr) [reply openoa/plant.py:2146]: Added try/catch here and pointed users to the Entr Alliance's py-entr repo.

**Review by RHammond2 — APPROVED** [review]

**Pull request merged**